diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x autom4te.cache -x config.cache -x config.log -x config.status -x config.h -x config.h.in -x ABOUT-NLS -x Makefile.in.in -x Makevars.template -x '*SlackBuild*' -x '*.egg-info' -x '*.class' -x '*.pyc' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x COPYING -x INSTALL -x compile -x config-ml.in -x config.guess -x config.sub -x depcomp -x elisp-comp -x install-sh -x ltmain.sh -x mdate-sh -x missing -x mkinstalldirs -x py-compile -x symlink-tree -x texinfo.tex -x ylwrap -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x intltool-extract -x intltool-extract.in -x intltool-merge -x intltool-merge.in -x intltool-update -x intltool-update.in -x '*.ps' origsrc/pdf417_enc.4.4/Makefile src/pdf417_enc.4.4/Makefile
--- origsrc/pdf417_enc.4.4/Makefile	2006-09-24 10:00:17.000000000 -0500
+++ src/pdf417_enc.4.4/Makefile	2007-07-31 23:28:16.750000000 -0500
@@ -14,7 +14,7 @@
 
 OBJS = main.o pdf417_enc.o tests.o
 LOBJS = pdf417_enc.o
-BINARIES = pdf417_enc libpdf417enc.a libpdf417enc.so lib_test
+BINARIES = pdf417_enc libpdf417enc.a cygpdf417enc.dll lib_test
 
 ifeq ($(UNAME),Other)
   OBJS += getopt.o
@@ -32,26 +32,26 @@
 
 all: pdf417_enc
 
-lib: libpdf417enc.so lib_test
+lib: libpdf417enc.a cygpdf417enc.dll lib_test
 
 pdf417_enc: $(OBJS)
-	$(LINK) $(LFLAGS) -o $@ $(OBJS)
+	$(LINK) -o $@ $(OBJS) $(LFLAGS)
 
-#libpdf417enc.a: $(LOBJS)
-#	ar cr $@ $(LOBJS)
-#	ranlib $@
+libpdf417enc.a: $(LOBJS)
+	ar cr $@ $(LOBJS)
+	ranlib $@
 
-libpdf417enc.so: $(LOBJS)
-	$(LINK) $(LFLAGS) -shared -o $@ $(LOBJS)
+cygpdf417enc.dll: $(LOBJS)
+	$(LINK) -shared -Wl,--enable-auto-image-base,--out-implib,libpdf417enc.dll.a -o $@ $(LOBJS) $(LFLAGS)
 
-lib_test: lib_test.o libpdf417enc.so
-	$(LINK) lib_test.o $(LFLAGS) -o $@ -L. -lpdf417enc
+lib_test: lib_test.o cygpdf417enc.dll
+	$(LINK) lib_test.o -o $@ -L. -lpdf417enc $(LFLAGS)
 
-lib_test2: lib_test2.o libpdf417enc.so
-	$(LINK) lib_test2.o $(LFLAGS) -o $@ -L. -lpdf417enc
+lib_test2: lib_test2.o cygpdf417enc.dll
+	$(LINK) lib_test2.o -o $@ -L. -lpdf417enc $(LFLAGS)
 
-lib_test_old: lib_test_old.o libpdf417enc.so
-	$(LINK) lib_test_old.o $(LFLAGS) -o $@ -L. -lpdf417enc
+lib_test_old: lib_test_old.o cygpdf417enc.dll
+	$(LINK) lib_test_old.o -o $@ -L. -lpdf417enc $(LFLAGS)
 
 clean:
 	rm -f lib_test.o $(OBJS) $(LOBJS) $(BINARIES)
