--- origsrc/pdf417_enc.4.4/Makefile	2006-09-24 10:00:17.000000000 -0500
+++ src/pdf417_enc.4.4/Makefile	2007-07-31 23:28:16.750000000 -0500
@@ -14,7 +14,7 @@
 
 OBJS = main.o pdf417_enc.o tests.o
 LOBJS = pdf417_enc.o
-BINARIES = pdf417_enc libpdf417enc.a libpdf417enc.so lib_test
+BINARIES = pdf417_enc libpdf417enc.a cygpdf417enc.dll lib_test
 
 ifeq ($(UNAME),Other)
   OBJS += getopt.o
@@ -32,26 +32,26 @@
 
 all: pdf417_enc
 
-lib: libpdf417enc.so lib_test
+lib: libpdf417enc.a cygpdf417enc.dll lib_test
 
 pdf417_enc: $(OBJS)
-	$(LINK) $(LFLAGS) -o $@ $(OBJS)
+	$(LINK) -o $@ $(OBJS) $(LFLAGS)
 
-#libpdf417enc.a: $(LOBJS)
-#	ar cr $@ $(LOBJS)
-#	ranlib $@
+libpdf417enc.a: $(LOBJS)
+	ar cr $@ $(LOBJS)
+	ranlib $@
 
-libpdf417enc.so: $(LOBJS)
-	$(LINK) $(LFLAGS) -shared -o $@ $(LOBJS)
+cygpdf417enc.dll: $(LOBJS)
+	$(LINK) -shared -Wl,--enable-auto-image-base,--out-implib,libpdf417enc.dll.a -o $@ $(LOBJS) $(LFLAGS)
 
-lib_test: lib_test.o libpdf417enc.so
-	$(LINK) lib_test.o $(LFLAGS) -o $@ -L. -lpdf417enc
+lib_test: lib_test.o cygpdf417enc.dll
+	$(LINK) lib_test.o -o $@ -L. -lpdf417enc $(LFLAGS)
 
-lib_test2: lib_test2.o libpdf417enc.so
-	$(LINK) lib_test2.o $(LFLAGS) -o $@ -L. -lpdf417enc
+lib_test2: lib_test2.o cygpdf417enc.dll
+	$(LINK) lib_test2.o -o $@ -L. -lpdf417enc $(LFLAGS)
 
-lib_test_old: lib_test_old.o libpdf417enc.so
-	$(LINK) lib_test_old.o $(LFLAGS) -o $@ -L. -lpdf417enc
+lib_test_old: lib_test_old.o cygpdf417enc.dll
+	$(LINK) lib_test_old.o -o $@ -L. -lpdf417enc $(LFLAGS)
 
 clean:
 	rm -f lib_test.o $(OBJS) $(LOBJS) $(BINARIES)
